# Домашнее задание

# Задание 1

Скачайте с сайта grouplens.org...movielens/ датасет любого размера. Определите какому фильму было выставлено больше всего оценок 5.0.

In [1]:
import pandas as pd

In [8]:
data = pd.read_csv('ratings.csv')
data.head()

userId  movieId  rating  timestamp
0       1        1     4.0  964982703
1       1        3     4.0  964981247
2       1        6     4.0  964982224
3       1       47     5.0  964983815
4       1       50     5.0  964982931

In [35]:
# проверяем, есть ли такие оценки
data.sort_values(by='rating', ascending=False).head()

userId  movieId  rating   timestamp
34031     232     3147     5.0  1218164903
50717     328     1080     5.0  1494210450
18166     115      487     5.0   944951378
18165     115      480     5.0   944953733
76157     479     3308     5.0  1039362558

In [40]:
# пишем проверку, где перввое значение это Id фильма, а второе колличество оценок 5.0
# где value_counts = распределение количества строк по значениям столбца
data [data ['rating'] == 5.0] ['movieId']. value_counts (). head (1)

318    153
Name: movieId, dtype: int64

Следствием чего является, что оценок 5.0 больше всего было выставлено фильму с Id: 318

# Задание 2

По данным файла power.csv посчитайте суммарное потребление стран Прибалтики (Латвия, Литва и Эстония) категорий 4, 12 и 21 за период с 2005 по 2010 года. Не учитывайте в расчетах отрицательные значения quantity.

In [42]:
data_w = pd.read_csv('power.csv')
data_w.head(10)

country  year  quantity  category
0  Austria  1996       5.0         1
1  Austria  1995      17.0         1
2  Belgium  2014       0.0         1
3  Belgium  2013       0.0         1
4  Belgium  2012      35.0         1
5  Belgium  2011      25.0         1
6  Belgium  2010      22.0         1
7  Belgium  2009      45.0         1
8  Czechia  1998       1.0         1
9  Czechia  1995       7.0         1

In [ ]:
Проверка существование этех стран в данных

In [70]:
#data_w [data_w ['country'] == 'Latvia']. value_counts (). head (1)

In [71]:
#data_w [data_w ['country'] == 'Lithuania']. value_counts (). head (1)

In [72]:
#data_w [data_w ['country'] == 'Estonia']. value_counts (). head (1)

Выделение этих стран в отдельный датасет

In [57]:
filt_country = data_w[ (data_w['country']=='Latvia') | (data_w['country']=='Lithuania')  | (data_w['country']=='Estonia')]
filt_country.head()

country  year  quantity  category
373  Lithuania  2014      39.0         1
374  Lithuania  2013     192.0         1
375  Lithuania  2012     196.0         1
376  Lithuania  2011     171.0         1
377  Lithuania  2010     158.0         1

Фильтруем по категориям 4,12,21

In [97]:
filt_country_by_category = filt_country[ (filt_country['category']==4) |  (filt_country['category']==12) |  (filt_country['category']==21)]
filt_country_by_category.head()

country  year  quantity  category
10617  Estonia  2008       1.0         4
10618  Estonia  2007       0.0         4
10619  Estonia  2006       0.0         4
10620  Estonia  2005       1.0         4
10621  Estonia  2004       1.0         4

In [87]:
len(filt_country_by_category)

1253

Фильтруем по требуемым годам с 2005 по 2010

In [150]:
filt_country_by_year = filt_country_by_category[ (filt_country_by_category ['year']>= 2005) & (filt_country_by_category ['year'] <= 2010)]
filt_country_by_year.head()


country  year  quantity  category
10617    Estonia  2008       1.0         4
10618    Estonia  2007       0.0         4
10619    Estonia  2006       0.0         4
10620    Estonia  2005       1.0         4
11156  Lithuania  2008       1.0         4

In [152]:
len(filt_country_by_year)

411

Фильтруем отрицательные значения quantity

In [153]:
filt_country_by_negative = filt_country_by_year [filt_country_by_year.quantity > 0 ]
filt_country_by_negative.head()

country  year  quantity  category
10617    Estonia  2008       1.0         4
10620    Estonia  2005       1.0         4
11156  Lithuania  2008       1.0         4
11158  Lithuania  2006       1.0         4
11159  Lithuania  2005       1.0         4

In [154]:
# проверка результата сортировки отрицательных значений
filt_country_by_negative.sort_values('quantity', ascending = True)

country  year  quantity  category
10617     Estonia  2008       1.0         4
26522   Lithuania  2006       1.0         4
26523   Lithuania  2005       1.0         4
103576  Lithuania  2010       1.0        12
104143  Lithuania  2008       1.0        12
...           ...   ...       ...       ...
248676  Lithuania  2006    4587.0        21
248675  Lithuania  2007    4613.0        21
248674  Lithuania  2008    4655.0        21
243152  Lithuania  2009    4689.0        21
248673  Lithuania  2009    4909.0        21

[347 rows x 4 columns]

In [155]:
# выберем как отдельный столбец
s_quantity = filt_country_by_negative['quantity']
s_quantity.head()

10617    1.0
10620    1.0
11156    1.0
11158    1.0
11159    1.0
Name: quantity, dtype: float64

In [156]:
s_quantity.sum()

240580.0

# Задание 3

Выберите страницу любого сайта с табличными данными. Импортируйте таблицы в pandas dataframe.
Примеры страниц (необязательно брать именно эти):
https://fortrader.org/quotes
www.finanz.ru...om-vremeni

In [181]:
pager_url = 'https://www.sports.ru/rfpl/table/'

idc = pd.read_html(pager_url, encoding='utf-8')
idc

[                      0       1
 0      Место проведения  Россия
 1  Последний победитель   Зенит,
     Unnamed: 0          Команда  М  В  Н  П  Заб  Проп   О
 0            1            Зенит  8  6  2  0   19     9  20
 1            2           Динамо  8  5  1  2   12     8  16
 2            3        Локомотив  8  4  4  0   12     6  16
 3            4             Сочи  8  5  0  3   12     6  15
 4            5            Рубин  8  4  2  2   13     8  14
 5            6        Краснодар  8  4  1  3   14     9  13
 6            7             ЦСКА  8  4  1  3   11     8  13
 7            8  Нижний Новгород  8  3  2  3   11    11  11
 8            9          Спартак  8  3  1  4    8     8  10
 9           10   Крылья Советов  8  3  1  4   10    11  10
 10          11            Ахмат  8  3  0  5    9    14   9
 11          12              Уфа  8  2  2  4   11    14   8
 12          13     Арсенал Тула  8  2  2  4   12    17   8
 13          14           Ростов  8  1  3  4   10    17   6
